Changelog v1.15 (#82)
meilisearch/engine-team: meilisearch-release-changelog/v1.15.0.md

# v1.15.0 release changelogs

<!-- The following line should ONLY be put PRE-release changelogs -->
> [!WARNING]
> Since this is a release candidate (RC), we do NOT recommend using it in a production environment. Is something not working as expected? We welcome [bug reports](https://github.com/meilisearch/meilisearch/issues/new/choose) and [feedback about new features](https://github.com/meilisearch/product/discussions).

> Meilisearch v1.15.0 introduces improves the indexing performances and dumpless experience.

Reviewer: Are they the things we want to promote?
-Dumpless upgrade is really niche to me (cf my comment under the section)
- performance improvements? about the new setting indexer?

We could promote: the new setting to Deactivate Typo Tolerance on Numbers and https://github.com/meilisearch/meilisearch/pull/5535

<!-- The following lines should NOT be put in the PRE-release changelogs -->
🧰 All official Meilisearch integrations (including SDKs, clients, and other tools) are compatible with this Meilisearch release. Integration deployment happens between 4 to 48 hours after a new version becomes available.

<!-- The following lines should NOT be put in the PRE-release changelogs -->
Some SDKs might not include all new features. Consult the project repository for detailed information. Is a feature you need missing from your chosen SDK? Create an issue letting us know you need it, or, for open-source karma points, open a PR implementing it (we'll love you for that ❤️).

# New features and updates 🔥

## Deactivate Typo Tolerance on Numbers

> TBD

Author: @ManyTheFish Would you mind describing your feature, please? The settings or the way to use it.

Done by @ManyTheFish in [#5494](https://github.com/meilisearch/meilisearch/pull/5494).

## Make it Possible to Rollback Dumpless Upgrades

> TBD

Author: Looks like there are some backtick issues in there :)

> TBD

Reviewer 2: fixed

> TBD

Reviewer: Honestly, I would not do a section for that, it's really niche, no OSS users asked for it. We develop it for the Cloud. For me it's like the snapshot compaction.
I would only put a line in "Other improvements".

It makes the changelogs less digest and hide features we want to promote like: https://github.com/meilisearch/meilisearch/pull/5535

Done by @dureuill in [#5523](https://github.com/meilisearch/meilisearch/pull/5523).

> 

Author: I will add https://github.com/meilisearch/meilisearch/pull/5535 to the list of main features!

> ## Other improvements

Reviewer: Is this the definitive order of the list below? I think we want to promote the most useful (for users!) first.

Reviewer 3: Reordered by putting:
1) new features/abilities
2) new informational fields
3) Error messages enhancements
4) experimental related changes

* Enhance filterable error messages by @CodeMan62 in [#5425](https://github.com/meilisearch/meilisearch/pull/5425)
* Improve error messages on embeddings dimension mismatch by @vuthanhtung2412 in [#5449](https://github.com/meilisearch/meilisearch/pull/5449)
* Support EC private key as SSL certificate by @HDT3213 in [#5471](https://github.com/meilisearch/meilisearch/pull/5471)
* Stop compacting the snapshots to speedup their generation by @dureuill in [#5498](https://github.com/meilisearch/meilisearch/pull/5498)
* Update network URL validation error message format to match expected by @CodeMan62 in [#5486](https://github.com/meilisearch/meilisearch/pull/5486)
* Add a new `batchCreationComplete` field in the batches stats by @dureuill in [#5488](https://github.com/meilisearch/meilisearch/pull/5488) and [#5530](https://github.com/meilisearch/meilisearch/pull/5530)

> * Expose the size of the status of the task queue in the prometheus metrics by @DanasFi in [#5512](https://github.com/meilisearch/meilisearch/pull/5512)

Reviewer: Can we link the associated experimental feature?
https://github.com/orgs/meilisearch/discussions/625

  -d '{
    "typoTolerance": {"disableOnNumbers": true}
  }'
```

Deactivating the typo tolerance on numbers can be useful when trying to reduce false positives, such as a query term `2024` returning results that include `2025` and `2004`. It may also improve indexing performance.

Done by @ManyTheFish in [#5494](https://github.com/meilisearch/meilisearch/pull/5494).

## Lexicographic string filters

This release allows you to filter strings lexicographically by enabling comparison operators (<, <=, >, >=, TO) on string values:
```
curl -X POST 'http://localhost:7700/indexes/movies/search' \
  -H 'Content-Type: application/json' \
  -d '{
    "filter": "release_date >= '2024-06'"
  }'
```

This new feature can be particularly useful when filtering human-readable dates.

Done by @dureuill in [#5535](https://github.com/meilisearch/meilisearch/pull/5535).

## Other improvements

* Allows cancelling an upgrade to a new Meilisearch version by rolling back all upgraded indexes by @dureuill in [#5523](https://github.com/meilisearch/meilisearch/pull/5523)
* Support EC private key as SSL certificate by @HDT3213 in [#5471](https://github.com/meilisearch/meilisearch/pull/5471)

> * Stop compacting snapshots to speed up their generation by @dureuill in [#5498](https://github.com/meilisearch/meilisearch/pull/5498)

Reviewer 2: ```suggestion
* When passing the relevant [CLI option](https://github.com/orgs/meilisearch/discussions/833), stop compacting snapshots, to speed up their generation by @dureuill and @irevoire  in [#5498](https://github.com/meilisearch/meilisearch/pull/5498) and [#5560](https://github.com/meilisearch/meilisearch/pull/5560)
```

* Add new `batchCreationComplete` field in the batches stats by @dureuill in [#5488](https://github.com/meilisearch/meilisearch/pull/5488) and [#5530](https://github.com/meilisearch/meilisearch/pull/5530)
* Add log field tracking time spent searching in the vector store by @Kerollmops in [#5525](https://github.com/meilisearch/meilisearch/pull/5525)
* Improve filterable error messages by @CodeMan62 in [#5425](https://github.com/meilisearch/meilisearch/pull/5425)
* Improve error messages on embeddings dimension mismatch by @vuthanhtung2412 in [#5449](https://github.com/meilisearch/meilisearch/pull/5449)
* Update `/network` URL validation error message format by @CodeMan62 in [#5486](https://github.com/meilisearch/meilisearch/pull/5486)
* Expose the task queue's status size in the Prometheus metrics by @DanasFi in [#5512](https://github.com/meilisearch/meilisearch/pull/5512)

# Fixes 🐞

* Fix `_matchesPosition` length calculation to improve client-side cropping by @shaokeyibb in [#5446](https://github.com/meilisearch/meilisearch/pull/5446)
* Fix `_geo` ranking rule by @HDT3213 in [#5487](https://github.com/meilisearch/meilisearch/pull/5487)

> 

Reviewer 2: ```suggestion
* Fix a panic in the search that could happen when looking for typos with a search prefix having more than 65k possible hits in the DB by @dureuill in [#5564](https://github.com/meilisearch/meilisearch/pull/5564)
* Make sure that passing `MEILI_EXPERIMENTAL_MAX_NUMBER_OF_BATCHED_TASKS` to 0 results in Meilisearch never processing any kind of task. By @irevoire in [#5565](https://github.com/meilisearch/meilisearch/pull/5565)
* Forbid value `0` for `maxTotalHits` in the index settings by @irevoire in [#5566](https://github.com/meilisearch/meilisearch/pull/5566)

```

## Lexicographic string filters

This release allows you to filter strings lexicographically by enabling comparison operators (<, <=, >, >=, TO) on string values:
```
curl -X POST 'http://localhost:7700/indexes/movies/search' \
  -H 'Content-Type: application/json' \
  -d '{
    "filter": "release_date >= '2024-06'"
  }'
```

This new feature can be particularly useful when filtering human-readable dates.

Done by @dureuill in [#5535](https://github.com/meilisearch/meilisearch/pull/5535).

## Other improvements

* Allows cancelling an upgrade to a new Meilisearch version by rolling back all upgraded indexes by @dureuill in [#5523](https://github.com/meilisearch/meilisearch/pull/5523)
* Support EC private key as SSL certificate by @HDT3213 in [#5471](https://github.com/meilisearch/meilisearch/pull/5471)
* When passing the relevant [CLI option](https://github.com/orgs/meilisearch/discussions/833), stop compacting snapshots, to speed up their generation by @dureuill and @irevoire in [#5498](https://github.com/meilisearch/meilisearch/pull/5498) and [#5560](https://github.com/meilisearch/meilisearch/pull/5560)

> * Add new `batchCreationComplete` field in the batches stats by @dureuill in [#5488](https://github.com/meilisearch/meilisearch/pull/5488) and [#5530](https://github.com/meilisearch/meilisearch/pull/5530)

Reviewer 2: ```suggestion
* Add new `batchStrategy` field in the batches stats by @dureuill in [#5488](https://github.com/meilisearch/meilisearch/pull/5488) and [#5530](https://github.com/meilisearch/meilisearch/pull/5530) and [#5588](https://github.com/meilisearch/meilisearch/pull/5588)
```

> [!WARNING]
> Since this is a release candidate (RC), we do NOT recommend using it in a production environment. Is something not working as expected? We welcome [bug reports](https://github.com/meilisearch/meilisearch/issues/new/choose) and [feedback about new features](https://github.com/meilisearch/product/discussions).

Meilisearch v1.15 adds a new typo tolerance setting, allowing you to disable typo tolerance for numbers. It also enables comparison operators for string filters.

<!-- The following lines should NOT be put in the PRE-release changelogs -->
🧰 All official Meilisearch integrations (including SDKs, clients, and other tools) are compatible with this Meilisearch release. Integration deployment happens between 4 to 48 hours after a new version becomes available.

<!-- The following lines should NOT be put in the PRE-release changelogs -->
Some SDKs might not include all new features. Consult the project repository for detailed information. Is a feature you need missing from your chosen SDK? Create an issue letting us know you need it, or, for open-source karma points, open a PR implementing it (we'll love you for that ❤️).

> # New features and updates 🔥

Author: I still need to add a line about our new `contact-email` CLI param and `MEILI_CONTACT_EMAIL` env var, as it is a breaking change, as I haven't implemented the timeout thing (yet?).

Author: Reverted by https://github.com/meilisearch/meilisearch/pull/5646

* Expose the size of the status of the task queue in the prometheus metrics by @DanasFi in [#5512](https://github.com/meilisearch/meilisearch/pull/5512) related to the [Prometheus Metrics experimental features](https://github.com/orgs/meilisearch/discussions/625)

# Fixes 🐞

* Fix `_matchesPosition` length calculation to improve client-side cropping by @shaokeyibb in [#5446](https://github.com/meilisearch/meilisearch/pull/5446)
* Fix the `_geo` ranking rule by @HDT3213 in [#5487](https://github.com/meilisearch/meilisearch/pull/5487)

# Misc

* Dependencies updates
  * Update the Rust version to 1.85.1 by @Kerollmops in [#5480](https://github.com/meilisearch/meilisearch/pull/5480)
  * Update charabia v0.9.3 by @ManyTheFish in [#5466](https://github.com/meilisearch/meilisearch/pull/5466)
  * Update the prometheus version to fix a vulnerability by @Kerollmops in [#5513](https://github.com/meilisearch/meilisearch/pull/5513)
* CIs and tests
  * Setup the Milestone CI to update the Ruleset by @Kerollmops in [#5444](https://github.com/meilisearch/meilisearch/pull/5444)

>   * Make the CI work with the GitHub Merge Queues Grouping by @Kerollmops in [#5445](https://github.com/meilisearch/meilisearch/pull/5445)

Reviewer 4: ```suggestion
  * Make the CI work with GitHub's merge queue grouping by @Kerollmops in [#5445](https://github.com/meilisearch/meilisearch/pull/5445)
```

  * Remove Bors references from the repository by @Kerollmops in [#5447](https://github.com/meilisearch/meilisearch/pull/5447)
  * Split unit tests for tasks by @shu-kitamura in [#5424](https://github.com/meilisearch/meilisearch/pull/5424)
* Misc

>   * Improve the process for dumpless upgrade to ensure compatibility by @irevoire in [#5395](https://github.com/meilisearch/meilisearch/pull/5395)

Reviewer 4: ```suggestion
  * Improve dumpless upgrade compatibility by @irevoire in [#5395](https://github.com/meilisearch/meilisearch/pull/5395)
```
